flo-runner/src/test/java/com/spotify/flo/context/FloRunnerTest.java

```diff
@@ -156,4 +175,81 @@ public void exceptionsExitNonZero() throws Exception {
 
     assertThat(status.get(), is(1));
   }
+
+  @Test
+  public void handleInvalidTerminationHookFactories() throws Exception {
+    TestTerminationHookFactory.injectHook(exceptionalTerminationHook);
+
+    AtomicInteger status = new AtomicInteger();
+    runTask(FOO_TASK).waitAndExit(status::set);
+
+    verify(exceptionalTerminationHook, times(1)).accept(0);
+
+    assertThat(status.get(), is(0));
+  }
+
+  @Test
+  public void testValidTerminationHook() {
+    final HookResult hookResult = new HookResult();
+
+    doAnswer(invocation -> {
+      Integer exitCode = invocation.getArgumentAt(0, Integer.class);
```

Comment:
You can use `ArgumentCaptor` to simplify this a bit. Just fyi.

Reply:
💲 good to know. I realized I can simplify it even more though, no need for captors either